### PR TITLE
Feature/improvements trigger

### PR DESCRIPTION
**Pull request by robYP**
Cleaned up / simplified "send_email_notification" function:
- added a new function "prepare_email_data"
- added EmailService for sending email

Slightly edited the error messages.

### CODE REVIEW

File: api/v1/trigger.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def get_event_route(event_id: int,
                     service: TriggerService = Depends(get_trigger_service)):
     route = service.get_event_route(event_id)
     if not route:
-        raise HTTPException(status_code=404, detail="event or route not found")
+        raise HTTPException(status_code=401, detail=f"Event {event_id} not found or has no route")
```
Comment (jonyig) [inline api/v1/trigger.py]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
這邊的錯誤是 Unauthorized 嗎？

File: services/trigger_service.py [hunk 2]
```diff
@@ -46,6 +44,8 @@ def get_event_content(self, event_id: int, user_id: int) -> Optional[str]:
             return None
 
         content = self.content_repository.get_content(event_id=event_id, language=user.language)
+        if not content:
+            content = self.content_repository.get_content(event_id=event_id , language="EN")
```
Comment (jonyig) [inline services/trigger_service.py]: 有辦法把打兩次 db ，縮小成打一次就好嗎？

File: api/v1/trigger.py (hunk 1 above)
Comment (robYP) [reply api/v1/trigger.py]: 錯誤status code 修正為400
commit：
86f49e4294ab13e615518683ccefb3af37533699

File: services/trigger_service.py (hunk 2 above)
Comment (robYP) [reply services/trigger_service.py]: 已修正
commit:
74be20006c0fae4bcd2613c7ef67a1fbb294dbe6

**Review by jonyig — APPROVED** [review]

**Pull request merged**